projects/storefrontapp/src/app/spartacus/spartacus-b2c-configuration.module.ts

```diff
@@ -6,6 +6,18 @@ import {
   mediaConfig,
   PWAModuleConfig,
 } from '@spartacus/storefront';
+import { environment } from '../../environments/environment';
+
+const defaultBaseSite = [
+  'electronics-spa',
+  'electronics',
+  'apparel-de',
+  'apparel-uk',
+  'apparel-uk-spa',
+];
+const baseSite = environment.epdVisualization
+  ? ['electronics-epdvisualization-spa'].concat(defaultBaseSite)
```

Comment:
When I run your feature `yarn start` with B2C (`$SPA_ENV=b2c,epd-visualisation`), then I get an error from backend: `Base site electronics-epdvisualization-spa doesn't exist`

Reply:
There is a typo in you comment above, it should be (`$SPA_ENV=b2c,epd-visualization`), but you are right the site has not been created, so there must be an issue with the sample data. I will look into it.

Reply:
I checked the response of the OCC endpoint `/basesites` and it looks that some sample data is missing in the test server https://api.cp96avkh5f-integrati1-d1-public.model-t.cc.commerce.ondemand.com:
- there is no EPD base site variation for electronics, e.g. `electronics-epdvisualisation-spa` (I highlighted all available basesites in red in the screenshot attached below)
- there is one missing URL pattern for `powertools-epdvisualisation-spa` - a regex that allows for recognition of URL `<storefront base url>/powertools-epdvisualisation-spa`. It causes page not to load, when I run Spartacus with the “automatic base site config” (https://sap.github.io/spartacus-docs/automatic-context-configuration), and launch the URL `/powertools-epdvisualisation-spa`. Please add appropriate URL pattern - you can take as an example the pattern from the basesite `powertools-spa` (I highlighted it in yellow in the screenshot below)

![image](https://user-images.githubusercontent.com/4001059/146363228-d9578abf-e065-4d1e-b3db-b75e643db6c8.png)

Reply:
There is no electronics-spaContentCatalog in my server, so I guess the spartacussampledataextension has not created it for some reason, which then would have meant that the electronics-epdvisualization-SpaContentCatalog was not created and then the site would not have been created.

![image](https://user-images.githubusercontent.com/52560012/146363917-6fe156ea-f1af-4aa1-b013-c83754854729.png)

Reply:
When you'll be fixing the addon to add the basesite `electronics-epdvisualization-spa`, please also make sure to add appropriate URL pattern for it allowing for Spartacus "[automatic basesite configuration](https://sap.github.io/spartacus-docs/automatic-context-configuration)".

Reply:
I don't think there is actually an issue with the `epdvisualizationspartacussampledata` extension. It looks like the database was not left in the expected state by the `spartacussampledata` extension which it has a declared dependency on (https://github.tools.sap/cx-commerce/epdvisualizationspartacussampledata/blob/cb02fd31e05c4ff202df5236ad8a3b75c81a668f/extensioninfo.xml#L9).

I have seen the sites created properly previously, so the first thing I will try is just running initialize again using the exact same build in case there was some issue with the upstream dependency tree.

Reply:
I have just started deploying to the CCv2 environment using the build that was deployed there previously. It will be down for the next hour or so.

Reply:
I read the comment about the missing regex after I had started the deployment.
I have made the change:
https://github.tools.sap/cx-commerce/epdvisualizationspartacussampledata/commit/4d69b84b2e4bcbd190b16a090edd3973f69f487b
But it won't be in the deployment that is in progress. I will do a build and deploy tomorrow which will include the change above.

Reply:
There was some kind of transient/intermittent issue affecting the creation of catalogs and sites. The missing site and catalog were created when the previously deployed build was redeployed a second time (with an initialize). I will see whether it is possible to find anything in the Kibana logs that explains what happened.

I will update this comment when the URL regex updates have been deployed to the CCv2 server.

Reply:
The catalogs and sites have been created successfully in the last two deploys.
The code that includes the new URL pattern has been deployed now.

I am not sure how to get the basesites API to be called by the application, but I invoked it manually and I get the following response now:

```json
{
   "baseSites" : [ {
      "channel" : "B2B",
      "defaultLanguage" : {
         "active" : true,
         "isocode" : "en",
         "name" : "English",
         "nativeName" : "English"
      },
      "defaultPreviewCategoryCode" : "1355",
      "defaultPreviewProductCode" : "2116282",
      "locale" : "en_GB",
      "name" : "Powertools Site",
      "stores" : [ {
         "createReturnProcessCode" : "return-process",
         "currencies" : [ {
            "active" : false,
            "isocode" : "USD",
            "name" : "US Dollar",
            "symbol" : "$"
         } ],
         "defaultCurrency" : {
            "active" : false,
            "isocode" : "USD",
            "name" : "US Dollar",
            "symbol" : "$"
         },
         "defaultDeliveryOrigin" : {
            "address" : {
               "country" : {
                  "isocode" : "US",
                  "name" : "United States"
               },
               "defaultAddress" : false,
               "formattedAddress" : "Brooklyn, 100 7th Avenue, New York, 11215",
               "id" : "8796126117911",
               "line1" : "Brooklyn",
               "line2" : "100 7th Avenue",
               "postalCode" : "11215",
               "shippingAddress" : false,
               "town" : "New York",
               "visibleInAddressBook" : true
            },
            "displayName" : "Warehouse East",
            "features" : {
            },
            "geoPoint" : {
               "latitude" : 40.7127,
               "longitude" : -74.0059
            },
            "name" : "pw_warehouse_e",
            "storeImages" : [ ]
         },
         "defaultLanguage" : {
            "active" : true,
            "isocode" : "en",
            "name" : "English",
            "nativeName" : "English"
         },
         "deliveryModes" : {
            "deliveryModes" : [ {
               "code" : "standard-gross",
               "description" : "3-5 business days",
               "name" : "Standard Delivery"
            }, {
               "code" : "free-standard-shipping"
            }, {
               "code" : "standard-net",
               "description" : "3-5 business days",
               "name" : "Standard Delivery"
            }, {
               "code" : "premium-net",
               "description" : "1-2 business days",
               "name" : "Premium Delivery"
            }, {
               "code" : "premium-gross",
               "description" : "1-2 business days",
               "name" : "Premium Delivery"
            } ]
         },
         "expressCheckoutEnabled" : false,
         "externalTaxEnabled" : false,
         "languages" : [ {
            "active" : true,
            "isocode" : "ja",
            "name" : "Japanese",
            "nativeName" : "日本語"
         }, {
            "active" : true,
            "isocode" : "en",
            "name" : "English",
            "nativeName" : "English"
         }, {
            "active" : true,
            "isocode" : "de",
            "name" : "German",
            "nativeName" : "Deutsch"
         }, {
            "active" : true,
            "isocode" : "zh",
            "name" : "Chinese",
            "nativeName" : "中文"
         } ],
         "maxRadiusForPosSearch" : 300.0,
         "name" : "Powertools Store",
         "paymentProvider" : "Mockup",
         "submitOrderProcessCode" : "order-process"
      } ],
      "theme" : "lambda",
      "uid" : "powertools",
      "urlEncodingAttributes" : [ "storefront", "language", "currency" ],
      "urlPatterns" : [ "(?i)^https?://[^/]+(/[^?]*)?\\?(.*\\&)?(site=powertools)(|\\&.*)$", "(?i)^https?://powertools\\.[^/]+(|/.*|\\?.*)$" ]
   }, {
      "channel" : "B2B",
      "defaultLanguage" : {
         "active" : true,
         "isocode" : "en",
         "name" : "English",
         "nativeName" : "English"
      },
      "defaultPreviewCatalogId" : "powertoolsProductCatalog",
      "defaultPreviewCategoryCode" : "1355",
      "defaultPreviewProductCode" : "2116282",
      "name" : "Spartacus powertools Site",
      "stores" : [ {
         "createReturnProcessCode" : "return-process",
         "currencies" : [ {
            "active" : false,
            "isocode" : "USD",
            "name" : "US Dollar",
            "symbol" : "$"
         } ],
         "defaultCurrency" : {
            "active" : false,
            "isocode" : "USD",
            "name" : "US Dollar",
            "symbol" : "$"
         },
         "defaultDeliveryOrigin" : {
            "address" : {
               "country" : {
                  "isocode" : "US",
                  "name" : "United States"
               },
               "defaultAddress" : false,
               "formattedAddress" : "Brooklyn, 100 7th Avenue, New York, 11215",
               "id" : "8796126117911",
               "line1" : "Brooklyn",
               "line2" : "100 7th Avenue",
               "postalCode" : "11215",
               "shippingAddress" : false,
               "town" : "New York",
               "visibleInAddressBook" : true
            },
            "displayName" : "Warehouse East",
            "features" : {
            },
            "geoPoint" : {
               "latitude" : 40.7127,
               "longitude" : -74.0059
            },
            "name" : "pw_warehouse_e",
            "storeImages" : [ ]
         },
         "defaultLanguage" : {
            "active" : true,
            "isocode" : "en",
            "name" : "English",
            "nativeName" : "English"
         },
         "deliveryModes" : {
            "deliveryModes" : [ {
               "code" : "standard-gross",
               "description" : "3-5 business days",
               "name" : "Standard Delivery"
            }, {
               "code" : "free-standard-shipping"
            }, {
               "code" : "standard-net",
               "description" : "3-5 business days",
               "name" : "Standard Delivery"
            }, {
               "code" : "premium-net",
               "description" : "1-2 business days",
               "name" : "Premium Delivery"
            }, {
               "code" : "premium-gross",
               "description" : "1-2 business days",
               "name" : "Premium Delivery"
            } ]
         },
         "expressCheckoutEnabled" : false,
         "externalTaxEnabled" : false,
         "languages" : [ {
            "active" : true,
            "isocode" : "ja",
            "name" : "Japanese",
            "nativeName" : "日本語"
         }, {
            "active" : true,
            "isocode" : "en",
            "name" : "English",
            "nativeName" : "English"
         }, {
            "active" : true,
            "isocode" : "de",
            "name" : "German",
            "nativeName" : "Deutsch"
         }, {
            "active" : true,
            "isocode" : "zh",
            "name" : "Chinese",
            "nativeName" : "中文"
         } ],
         "maxRadiusForPosSearch" : 300.0,
         "name" : "Powertools Store",
         "paymentProvider" : "Mockup",
         "submitOrderProcessCode" : "order-process"
      } ],
      "theme" : "sparta",
      "uid" : "powertools-spa",
      "urlEncodingAttributes" : [ "storefront", "language", "currency" ],
      "urlPatterns" : [ "(?i)^https?://[^/]+(/[^?]*)?\\?(.*\\&)?(site=powertools-spa)(|\\&.*)$", "(?i)^https?://powertools-spa\\.[^/]+(|/.*|\\?.*)$", "(?i)^https?://api\\.hybrisdev\\.com(:[\\d]+)?/rest/.*$", "(?i)^https?://localhost(:[\\d]+)?/rest/.*$", "(?i)^https?://[^/]+/powertools-spa(|/.*|\\?.*|#.*)$" ]
   }, {
      "channel" : "B2B",
      "defaultLanguage" : {
         "active" : true,
         "isocode" : "en",
         "name" : "English",
         "nativeName" : "English"
      },
      "defaultPreviewCatalogId" : "powertoolsProductCatalog",
      "defaultPreviewCategoryCode" : "1355",
      "defaultPreviewProductCode" : "2116282",
      "name" : "Spartacus EPD Visualization Powertools Site",
      "stores" : [ {
         "createReturnProcessCode" : "return-process",
         "currencies" : [ {
            "active" : false,
            "isocode" : "USD",
            "name" : "US Dollar",
            "symbol" : "$"
         } ],
         "defaultCurrency" : {
            "active" : false,
            "isocode" : "USD",
            "name" : "US Dollar",
            "symbol" : "$"
         },
         "defaultDeliveryOrigin" : {
            "address" : {
               "country" : {
                  "isocode" : "US",
                  "name" : "United States"
               },
               "defaultAddress" : false,
               "formattedAddress" : "Brooklyn, 100 7th Avenue, New York, 11215",
               "id" : "8796126117911",
               "line1" : "Brooklyn",
               "line2" : "100 7th Avenue",
               "postalCode" : "11215",
               "shippingAddress" : false,
               "town" : "New York",
               "visibleInAddressBook" : true
            },
            "displayName" : "Warehouse East",
            "features" : {
            },
            "geoPoint" : {
               "latitude" : 40.7127,
               "longitude" : -74.0059
            },
            "name" : "pw_warehouse_e",
            "storeImages" : [ ]
         },
         "defaultLanguage" : {
            "active" : true,
            "isocode" : "en",
            "name" : "English",
            "nativeName" : "English"
         },
         "deliveryModes" : {
            "deliveryModes" : [ {
               "code" : "standard-gross",
               "description" : "3-5 business days",
               "name" : "Standard Delivery"
            }, {
               "code" : "free-standard-shipping"
            }, {
               "code" : "standard-net",
               "description" : "3-5 business days",
               "name" : "Standard Delivery"
            }, {
               "code" : "premium-net",
               "description" : "1-2 business days",
               "name" : "Premium Delivery"
            }, {
               "code" : "premium-gross",
               "description" : "1-2 business days",
               "name" : "Premium Delivery"
            } ]
         },
         "expressCheckoutEnabled" : false,
         "externalTaxEnabled" : false,
         "languages" : [ {
            "active" : true,
            "isocode" : "ja",
            "name" : "Japanese",
            "nativeName" : "日本語"
         }, {
            "active" : true,
            "isocode" : "en",
            "name" : "English",
            "nativeName" : "English"
         }, {
            "active" : true,
            "isocode" : "de",
            "name" : "German",
            "nativeName" : "Deutsch"
         }, {
            "active" : true,
            "isocode" : "zh",
            "name" : "Chinese",
            "nativeName" : "中文"
         } ],
         "maxRadiusForPosSearch" : 300.0,
         "name" : "Powertools Store",
         "paymentProvider" : "Mockup",
         "submitOrderProcessCode" : "order-process"
      } ],
      "theme" : "Santorini",
      "uid" : "powertools-epdvisualization-spa",
      "urlEncodingAttributes" : [ "storefront", "language" ],
      "urlPatterns" : [ "(?i)^https?://[^/]+(/[^?]*)?\\?(.*\\&)?(site=powertools-epdvisualization-spa)(|\\&.*)$", "(?i)^https?://powertools-epdvisualization-spa\\.[^/]+(|/.*|\\?.*)$", "(?i)^https?://api\\.hybrisdev\\.com(:[\\d]+)?/rest/.*$", "(?i)^https?://localhost(:[\\d]+)?/rest/.*$", "(?i)^https?://[^/]+/powertools-epdvisualization-spa(|/.*|\\?.*|#.*)$" ]
   }, {
      "channel" : "B2C",
      "defaultLanguage" : {
         "active" : true,
         "isocode" : "en",
         "name" : "English",
         "nativeName" : "English"
      },
      "defaultPreviewCatalogId" : "electronicsProductCatalog",
      "defaultPreviewCategoryCode" : "575",
      "defaultPreviewProductCode" : "2053367",
      "locale" : "en_GB",
      "name" : "Electronics Site",
      "stores" : [ {
         "createReturnProcessCode" : "return-process",
         "currencies" : [ {
            "active" : false,
            "isocode" : "JPY",
            "name" : "Japanese Yen",
            "symbol" : "¥"
         }, {
            "active" : false,
            "isocode" : "USD",
            "name" : "US Dollar",
            "symbol" : "$"
         } ],
         "defaultCurrency" : {
            "active" : false,
            "isocode" : "USD",
            "name" : "US Dollar",
            "symbol" : "$"
         },
         "defaultDeliveryOrigin" : {
            "address" : {
               "country" : {
                  "isocode" : "JP",
                  "name" : "Japan"
               },
               "defaultAddress" : false,
               "formattedAddress" : "1-5-16 Takadanobaba, Tokio, 169-0075",
               "id" : "8796128149527",
               "line2" : "1-5-16 Takadanobaba",
               "phone" : "+813 3201 8968",
               "postalCode" : "169-0075",
               "shippingAddress" : false,
               "town" : "Tokio",
               "visibleInAddressBook" : true
            },
            "displayName" : "Warehouse East",
            "features" : {
            },
            "geoPoint" : {
               "latitude" : 35.7116188,
               "longitude" : 139.7082147
            },
            "name" : "warehouse_e",
            "storeImages" : [ ]
         },
         "defaultLanguage" : {
            "active" : true,
            "isocode" : "en",
            "name" : "English",
            "nativeName" : "English"
         },
         "deliveryModes" : {
            "deliveryModes" : [ {
               "code" : "premium-gross",
               "description" : "1-2 business days",
               "name" : "Premium Delivery"
            }, {
               "code" : "free-standard-shipping"
            }, {
               "code" : "standard-net",
               "description" : "3-5 business days",
               "name" : "Standard Delivery"
            }, {
               "code" : "standard-gross",
               "description" : "3-5 business days",
               "name" : "Standard Delivery"
            }, {
               "code" : "premium-net",
               "description" : "1-2 business days",
               "name" : "Premium Delivery"
            }, {
               "code" : "pickup"
            } ]
         },
         "expressCheckoutEnabled" : true,
         "externalTaxEnabled" : false,
         "languages" : [ {
            "active" : true,
            "isocode" : "ja",
            "name" : "Japanese",
            "nativeName" : "日本語"
         }, {
            "active" : true,
            "isocode" : "en",
            "name" : "English",
            "nativeName" : "English"
         }, {
            "active" : true,
            "isocode" : "de",
            "name" : "German",
            "nativeName" : "Deutsch"
         }, {
            "active" : true,
            "isocode" : "zh",
            "name" : "Chinese",
            "nativeName" : "中文"
         } ],
         "maxRadiusForPosSearch" : 300.0,
         "name" : "Electronics Store",
         "paymentProvider" : "Mockup",
         "submitOrderProcessCode" : "order-process"
      } ],
      "theme" : "alpha",
      "uid" : "electronics",
      "urlEncodingAttributes" : [ "storefront", "language" ],
      "urlPatterns" : [ "(?i)^https?://[^/]+(/[^?]*)?\\?(.*\\&)?(site=electronics)(|\\&.*)$", "(?i)^https?://electronics\\.[^/]+(|/.*|\\?.*)$", "(?i)^https?://api\\.hybrisdev\\.com(:[\\d]+)?/rest/.*$", "(?i)^https?://localhost(:[\\d]+)?/rest/.*$" ]
   }, {
      "channel" : "B2C",
      "defaultLanguage" : {
         "active" : true,
         "isocode" : "en",
         "name" : "English",
         "nativeName" : "English"
      },
      "defaultPreviewCatalogId" : "electronicsProductCatalog",
      "defaultPreviewCategoryCode" : "575",
      "defaultPreviewProductCode" : "2053367",
      "name" : "Spartacus Electronics Site",
      "stores" : [ {
         "createReturnProcessCode" : "return-process",
         "currencies" : [ {
            "active" : false,
            "isocode" : "JPY",
            "name" : "Japanese Yen",
            "symbol" : "¥"
         }, {
            "active" : false,
            "isocode" : "USD",
            "name" : "US Dollar",
            "symbol" : "$"
         } ],
         "defaultCurrency" : {
            "active" : false,
            "isocode" : "USD",
            "name" : "US Dollar",
            "symbol" : "$"
         },
         "defaultDeliveryOrigin" : {
            "address" : {
               "country" : {
                  "isocode" : "JP",
                  "name" : "Japan"
               },
               "defaultAddress" : false,
               "formattedAddress" : "1-5-16 Takadanobaba, Tokio, 169-0075",
               "id" : "8796128149527",
               "line2" : "1-5-16 Takadanobaba",
               "phone" : "+813 3201 8968",
               "postalCode" : "169-0075",
               "shippingAddress" : false,
               "town" : "Tokio",
               "visibleInAddressBook" : true
            },
            "displayName" : "Warehouse East",
            "features" : {
            },
            "geoPoint" : {
               "latitude" : 35.7116188,
               "longitude" : 139.7082147
            },
            "name" : "warehouse_e",
            "storeImages" : [ ]
         },
         "defaultLanguage" : {
            "active" : true,
            "isocode" : "en",
            "name" : "English",
            "nativeName" : "English"
         },
         "deliveryModes" : {
            "deliveryModes" : [ {
               "code" : "premium-gross",
               "description" : "1-2 business days",
               "name" : "Premium Delivery"
            }, {
               "code" : "free-standard-shipping"
            }, {
               "code" : "standard-net",
               "description" : "3-5 business days",
               "name" : "Standard Delivery"
            }, {
               "code" : "standard-gross",
               "description" : "3-5 business days",
               "name" : "Standard Delivery"
            }, {
               "code" : "premium-net",
               "description" : "1-2 business days",
               "name" : "Premium Delivery"
            }, {
               "code" : "pickup"
            } ]
         },
         "expressCheckoutEnabled" : true,
         "externalTaxEnabled" : false,
         "languages" : [ {
            "active" : true,
            "isocode" : "ja",
            "name" : "Japanese",
            "nativeName" : "日本語"
         }, {
            "active" : true,
            "isocode" : "en",
            "name" : "English",
            "nativeName" : "English"
         }, {
            "active" : true,
            "isocode" : "de",
            "name" : "German",
            "nativeName" : "Deutsch"
         }, {
            "active" : true,
            "isocode" : "zh",
            "name" : "Chinese",
            "nativeName" : "中文"
         } ],
         "maxRadiusForPosSearch" : 300.0,
         "name" : "Electronics Store",
         "paymentProvider" : "Mockup",
         "submitOrderProcessCode" : "order-process"
      } ],
      "theme" : "sparta",
      "uid" : "electronics-spa",
      "urlEncodingAttributes" : [ "storefront", "language", "currency" ],
      "urlPatterns" : [ "(?i)^https?://[^/]+(/[^?]*)?\\?(.*\\&)?(site=electronics-spa)(|\\&.*)$", "(?i)^https?://electronics-spa\\.[^/]+(|/.*|\\?.*)$", "(?i)^https?://api\\.hybrisdev\\.com(:[\\d]+)?/rest/.*$", "(?i)^https?://localhost(:[\\d]+)?/rest/.*$", "(?i)^https?://[^/]+/electronics-spa(|/.*|\\?.*|#.*)$", "(?i)^https?://localhost(:[\\d]+)?/?$" ]
   }, {
      "channel" : "B2C",
      "defaultLanguage" : {
         "active" : true,
         "isocode" : "en",
         "name" : "English",
         "nativeName" : "English"
      },
      "defaultPreviewCatalogId" : "electronicsProductCatalog",
      "defaultPreviewCategoryCode" : "575",
      "defaultPreviewProductCode" : "2053367",
      "name" : "Spartacus EPD Visualization Electronics Site",
      "stores" : [ {
         "createReturnProcessCode" : "return-process",
         "currencies" : [ {
            "active" : false,
            "isocode" : "JPY",
            "name" : "Japanese Yen",
            "symbol" : "¥"
         }, {
            "active" : false,
            "isocode" : "USD",
            "name" : "US Dollar",
            "symbol" : "$"
         } ],
         "defaultCurrency" : {
            "active" : false,
            "isocode" : "USD",
            "name" : "US Dollar",
            "symbol" : "$"
         },
         "defaultDeliveryOrigin" : {
            "address" : {
               "country" : {
                  "isocode" : "JP",
                  "name" : "Japan"
               },
               "defaultAddress" : false,
               "formattedAddress" : "1-5-16 Takadanobaba, Tokio, 169-0075",
               "id" : "8796128149527",
               "line2" : "1-5-16 Takadanobaba",
               "phone" : "+813 3201 8968",
               "postalCode" : "169-0075",
               "shippingAddress" : false,
               "town" : "Tokio",
               "visibleInAddressBook" : true
            },
            "displayName" : "Warehouse East",
            "features" : {
            },
            "geoPoint" : {
               "latitude" : 35.7116188,
               "longitude" : 139.7082147
            },
            "name" : "warehouse_e",
            "storeImages" : [ ]
         },
         "defaultLanguage" : {
            "active" : true,
            "isocode" : "en",
            "name" : "English",
            "nativeName" : "English"
         },
         "deliveryModes" : {
            "deliveryModes" : [ {
               "code" : "premium-gross",
               "description" : "1-2 business days",
               "name" : "Premium Delivery"
            }, {
               "code" : "free-standard-shipping"
            }, {
               "code" : "standard-net",
               "description" : "3-5 business days",
               "name" : "Standard Delivery"
            }, {
               "code" : "standard-gross",
               "description" : "3-5 business days",
               "name" : "Standard Delivery"
            }, {
               "code" : "premium-net",
               "description" : "1-2 business days",
               "name" : "Premium Delivery"
            }, {
               "code" : "pickup"
            } ]
         },
         "expressCheckoutEnabled" : true,
         "externalTaxEnabled" : false,
         "languages" : [ {
            "active" : true,
            "isocode" : "ja",
            "name" : "Japanese",
            "nativeName" : "日本語"
         }, {
            "active" : true,
            "isocode" : "en",
            "name" : "English",
            "nativeName" : "English"
         }, {
            "active" : true,
            "isocode" : "de",
            "name" : "German",
            "nativeName" : "Deutsch"
         }, {
            "active" : true,
            "isocode" : "zh",
            "name" : "Chinese",
            "nativeName" : "中文"
         } ],
         "maxRadiusForPosSearch" : 300.0,
         "name" : "Electronics Store",
         "paymentProvider" : "Mockup",
         "submitOrderProcessCode" : "order-process"
      } ],
      "theme" : "Santorini",
      "uid" : "electronics-epdvisualization-spa",
      "urlEncodingAttributes" : [ "storefront", "language" ],
      "urlPatterns" : [ "(?i)^https?://[^/]+(/[^?]*)?\\?(.*\\&)?(site=electronics-epdvisualization-spa)(|\\&.*)$", "(?i)^https?://electronics-epdvisualization-spa\\.[^/]+(|/.*|\\?.*)$", "(?i)^https?://api\\.hybrisdev\\.com(:[\\d]+)?/rest/.*$", "(?i)^https?://localhost(:[\\d]+)?/rest/.*$", "(?i)^https?://[^/]+/electronics-epdvisualization-spa(|/.*|\\?.*|#.*)$" ]
   } ]
```

Reply:
> I am not sure how to get the basesites API to be called by the application

Spartacus will call this endpoint before app starts when no config is provided for the: `context.baseSite`. So to enable the automatic basesite config (fetched from OCC), you ideally would need to remove (comment out) manually any `context` config provided in the app.

In our monorepo it means commenting out a few lines from `spartacus-[b2b|b2c]-configuration.module.ts` (depending whether you're running b2b|b2c variation of the storefrojnt)